central/node/datastore/store/postgres/store.go

```diff
@@ -860,6 +872,10 @@ func (s *storeImpl) retryableGetMany(ctx context.Context, ids []string) ([]*stor
 	resultsByID := make(map[string]*storage.Node)
 	for _, id := range ids {
 		msg, found, err := s.getFullNode(ctx, tx, id)
+		// No changes are made to the database, so COMMIT or ROLLBACK have same effect.
+		if err := tx.Commit(ctx); err != nil {
```

Comment:
This seems like we don't want to commit here. We have the commit below at the bottom

Reply:
CI has answered this question.  Commit closes the transaction so next call fails.

Reply:
Ah, the tx needs to be committed only if `err != nil`.